src/libgrate/matrix.c

```diff
@@ -143,7 +143,7 @@ void mat4_perspective(struct mat4 *m, float fov, float aspect,
 	m->xx = cotangent / aspect;
 	m->yy = cotangent;
 	m->zz = -(far + near) / depth;
-	m->zw = -1.0f;
-	m->wz = -2.0f * near * far / depth;
+	m->zw = -2.0f * near * far / depth;
+	m->wz = -1.0f;
```

Comment:
Hmm, this looks like we need the transpose of what something like [GLM would produce](https://github.com/g-truc/glm/blob/e8962dda2a2652a6a97432449681a4366876fc84/glm/gtc/matrix_transform.inl#L274)... Perhaps we should just switch to using GLM for the computations, and make our upload-code transpose instead (or perform matrix multiplications the other way around in the shader)?